src/test/java/org/datadog/jmxfetch/TestLogInitialization.java

```diff
@@ -0,0 +1,195 @@
+import javax.management.ObjectName;
+import javax.management.remote.JMXServiceURL;
+import org.junit.After;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static com.google.common.base.StandardSystemProperty.JAVA_CLASS_PATH;
+import static com.google.common.base.StandardSystemProperty.PATH_SEPARATOR;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+public class TestLogInitialization {
+    public static final Callable<AppConfig> LOCAL_CONFIG = new Callable<AppConfig>() {
+        public AppConfig call() throws Exception {
+            return AppConfig.create(
+                    ImmutableList.of("org/datadog/jmxfetch/dd-java-agent-jmx.yaml"),
+                    Collections.<String>emptyList(),
+                    (int) TimeUnit.SECONDS.toMillis(30),
+                    (int) TimeUnit.SECONDS.toMillis(30),
+                    Collections.<String, String>emptyMap(),
+                    "console",
+                    "System.err",
+                    "DEBUG");
+        }
+    };
+
+    public static final Callable<AppConfig> REMOTE_CONFIG = new Callable<AppConfig>() {
+        public AppConfig call() throws Exception {
+            return AppConfig.create(
+                    ImmutableList.of("org/datadog/jmxfetch/remote-jmx.yaml"),
+                    Collections.<String>emptyList(),
+                    (int) TimeUnit.SECONDS.toMillis(30),
+                    (int) TimeUnit.SECONDS.toMillis(30),
+                    Collections.<String, String>emptyMap(),
+                    "console",
+                    "System.err",
+                    "DEBUG");
+        }
+    };
+
+    @After
+    public void unregisterLockingMBean() {
+        try {
+        ManagementFactory.getPlatformMBeanServer()
+                .unregisterMBean(new ObjectName("org.datadog.jmxfetch.log_init_test:type=TriggeringMBean"));
+
+        } catch (Exception e) {
+            // Ignore
+        }
+    }
+
+    @Test
+    public void testLocalUsageDoesNotInitalizeLogManager() throws Exception {
+        CountDownLatch latch = registerLockingMBean();
+
+        TrackingClassLoader classLoader = new TrackingClassLoader();
+
+        final AtomicReference<Exception> errored = runInThread(classLoader, "LOCAL_CONFIG");
+
+        latch.await(15, TimeUnit.SECONDS);
+
+        assertNull(errored.get());
+        assertTrue(classLoader.classLoaded(LocalConnection.class.getName()));
+        assertFalse(classLoader.classLoaded(JMXServiceURL.class.getName()));
+    }
+
+    @Test
+    public void testRemoteUsageDoesInitalizeLogManager() throws Exception {
+        registerLockingMBean();
+
+        TrackingClassLoader classLoader = new TrackingClassLoader();
+
+        runInThread(classLoader, "REMOTE_CONFIG");
+
+        // We don't know how long until the error triggers.  No good way to verify.
```

Comment:
@olivielpeau the error is caught by `App.run`, so it never bubbles up.  I could refactor to block until that class is loaded as you suggested, but I'm not sure it will make a huge difference.  Is that important to you?

Reply:
Not that important to me, as long as the test isn't flaky 👍